classes/Collector.js

```diff
@@ -0,0 +1,217 @@
+import util from 'util';
+import moment from 'moment-timezone';
+import Q from 'q';
+import Promise from 'bluebird';
+import log from 'common/log';
+import BookModel from 'models/Book';
+import AuthorModel from 'models/Author';
+
+/**
+@param { Object } book - 保存用に正規化された書籍のデータ
+@param { Function } done - 完了時に呼ばれるコールバック関数
+**/
+function saveBook(book, done) {
+  // 必須パラメータの検証
+  if (typeof book !== 'object' || util.isArray(book)) { throw new Error('saveBook method required Object parametor.'); }
+  if (typeof done !== 'function' || !done) { throw new Error('saveBook method required Function parametor.'); }
+
+  // 必須項目の検証
+  if (!book.ASIN) { throw new Error('book param required ASIN property.'); }
+  if (!book.author) { throw new Error('book param required author property.'); }
+  if (!book.title) { throw new Error('book param required title property.'); }
+
+  const conditions = {
+    ASIN: book.ASIN,
+  };
+  const BookList = this._Model;
+
+  BookList.findOne(conditions, (err, existBook)=> {
+    // 既に書籍が存在していたらエラーハンドリングに回す
+    if (err) {
+      return done(err);
+    }
+    if (existBook) {
+      log.warn.info('登録済みの書籍:' + existBook.title);
+      return done();
+    }
+
+    const initialLibrarianTime = moment('2000-01-01');
+    const newBook = new BookList({
```

Comment:
__[jshint]__
*  'const' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).